### DISCUSSION TITLE
修改目录页和摘要页的页眉内容

### DISCUSSION THREAD

**Discussion by pixas** [sjtug/SJTUThesis#953] [提问]
您好，目前的摘要页
![image](https://github.com/sjtug/SJTUThesis/assets/58799127/f1614bd7-ca15-4769-86bf-f366969b07bd)
中的页眉包含了“目录”，我想要去掉这个内容，使其只保留左边的“上海交通大学学位论文”。
我只想对目录页、摘要页做这个操作，而不影响其他页面。我要如何实现呢？谢谢！

**Comment by LogCreative** [comment]
这不是[《撰写规范》](https://jwc.sjtu.edu.cn/__local/4/FF/C5/678CA7D260DEED00B4C37528646_67B1672C_604F2.pdf)的强制要求，不建议作出这种修改。

**Comment by pixas** [comment]
您好，谢谢您的提醒，但是我在提交图书馆专家论文审核的时候被审核专家提出了如下修改建议：
![image](https://github.com/sjtug/SJTUThesis/assets/58799127/9da66495-917b-4162-a6f5-691309bad4c0)
其中的第三条要求我修改。
我参考了学校提供的word模板
![image](https://github.com/sjtug/SJTUThesis/assets/58799127/083db82c-c2f2-4aae-ae86-b6fc2dc4912b)

请问这种情况下，为了满足图书馆专家的需求我要如何修改呢？谢谢您！

**Reply by LogCreative** [reply]
我们核实一下。

**Comment by LogCreative** [comment]
只删除目录页和摘要页的页眉章节内容方法如下：

```latex
{

\ExplSyntaxOn
\bool_if:NTF \g__sjtu_twoside_bool
{
    \fancyhead [ LE ]     { }
    \fancyhead [ RO ]     { }
}
{
    \fancyhead [ R ] { }
}
\ExplSyntaxOff

% 摘要
\input{contents/abstract}

% 目录
\tableofcontents
\clearpage

}
```

**Reply by jbwenjoy** [reply]
您好！

这种方式对摘要、目录和符号对照表有效，但文末的大摘要似乎无法通过这种方式删除章节标题页眉

<img width="446" alt="image" src="https://github.com/sjtug/SJTUThesis/assets/71893666/49efbc9b-c54d-4914-bb91-dcc411f2f7fb">

我是这样写的，不太清楚问题出在哪里
```
% in file main.tex
...
% 学士学位论文要求在最后有一个大摘要，单独编页码
{
    % 只删除大摘要的页眉章节标题
    \ExplSyntaxOn
    \bool_if:NTF \g__sjtu_twoside_bool
    {
        \fancyhead [ LE ]     { }
        \fancyhead [ RO ]     { }
    }
    {
        \fancyhead [ R ] { }
    }
    \ExplSyntaxOff

    % 大摘要
    \input{contents/digest}

    \clearpage
}
...
```
请问又应当如何处理呢？

非常感谢您！

**Reply by LogCreative** [reply]
这种情况清空大摘要的名称即可。

```latex
\sjtusetup{name/digest={}}
```

**Reply by dzx-dzx** [reply]
不知道具体原理是什么，但是原来的方法稍加改动也可以去掉页眉:
```latex
\input{contents/acknowledgements}
% 后面不再用大括号包裹
\clearpage % 把这个命令调整到前面

\ExplSyntaxOn
\bool_if:NTF \g__sjtu_twoside_bool
{
    \fancyhead [ LE ]     { }
    \fancyhead [ RO ]     { }
}
{
    \fancyhead [ R ] {  }
}
\ExplSyntaxOff
\input{contents/digest}
```

**Reply by LogCreative** [reply]
因为页眉和页脚是在整个页面主体部分最后渲染出来的，所以使用 `\clearpage` 可以显式地结束该页使得设置生效的区域有明确的边界。

**Comment by pixas** [comment]
谢谢您，我看似乎英文的摘要页
![image](https://github.com/sjtug/SJTUThesis/assets/58799127/e4c79d43-4087-4845-804f-91fe4ce95d31)
位于了右上角，我向调整到左上角的位置，请问要如何实现呢？感谢您！

**Reply by LogCreative** [reply]
你可能使用了 `openany` 选项，应当使用 `oneside`。

**Reply by pixas** [reply]
谢谢您，成功了
